File: percent.py
print("\n")
started_summ = int(input("Введите начальную сумму: "))
percent = float(input("Введите процент вклада: "))
capitalization = 0
mounths = int(input("Введите срок вклада в месяцах: "))
added_sum = int(input("Введите на какую сумму будете ежемесячно пополнять вклад: "))
end_summ = started_summ
print("\n")

def count(started_summ,percent,capitalization,mounths,added_sum,end_summ):
    mounth = 0
    full_period_percent = 0
    while mounth < mounths:
        capitalization = (started_summ * (percent / 100)) / 12
        end_summ = round(started_summ + capitalization, 0)
        full_period_percent += capitalization
        capitalization = 0
        started_summ = end_summ + added_sum
        mounth += 1
        print(f"В конце {mounth} месяца:" ,started_summ)
        print(f"Общий % = {round(full_period_percent, 2)} руб. \n")
    return end_summ

count(started_summ, percent, capitalization, mounths, added_sum, end_summ)

File: rust_index/src/percent.rs
pub fn count_sum(
    mut started_sum: f32,
    mut capitalization: f32,
    mut end_sum: f32,
    percent: f32,
    mounths: i32,
    added_sum: f32,
) -> f32 {
    let mut mounth: i32 = 0;
    let mut full_period_percent: f32 = 0.0;
    while mounth < mounths {
        capitalization = started_sum * (percent / 100.00) / 12.00;
        end_sum = started_sum + capitalization;
        full_period_percent = full_period_percent + capitalization;
        capitalization = 0.0;
        started_sum = end_sum + added_sum;
        mounth += 1;
        println!("В конце {} месяца: {} руб.", mounth, started_sum);
        println!("Общий %: {} руб. \n", full_period_percent);
    }
    return started_sum;
}
